gift-certificates-app/service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -2,46 +2,53 @@
 
 import com.epam.esm.bean.Tag;
 import com.epam.esm.dao.TagDAO;
+import com.epam.esm.dto.TagDTO;
+import com.epam.esm.mapper.TagMapper;
 import com.epam.esm.service.TagService;
-import com.epam.esm.validator.TagCheck;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 public class TagServiceImpl implements TagService {
 
     private final TagDAO tagDAO;
-    private final TagCheck tagCheck;
+    private final TagMapper tagMapper;
 
     @Autowired
-    public TagServiceImpl(TagDAO tagDAO, TagCheck tagCheck) {
+    public TagServiceImpl(TagDAO tagDAO, TagMapper tagMapper) {
         this.tagDAO = tagDAO;
-        this.tagCheck = tagCheck;
+        this.tagMapper = tagMapper;
     }
 
     @Override
     @Transactional
```

Comment:
you also have @Transactional for TagDAO class, it looks like transaction in the transaction
would be better to have @Transactional only in the service layer for each method (not for the whole class)

Reply:
+ I have deleted @Transactional from @Repository